### ISSUE TITLE
[FEA] Support Negative upper bound on current row in windowing

### ISSUE THREAD

**Issue by kuhushukla** [NVIDIA/spark-rapids#11801]
**Is your feature request related to a problem? Please describe.**
```
Exec <WindowExec> cannot run on GPU because not all expressions can be replaced
    !Expression <SpecifiedWindowFrame> specifiedwindowframe(my_range_frame, -12345, -1) cannot run on GPU because Upper-bounds behind current row is not supported. Found: -1
```
**Describe the solution you'd like**
Support upper bounds that can go behind the current row. 

**Describe alternatives you've considered**
None

**Additional context**
Customer query

**Comment by revans2** [comment]
@kuhushukla which window operations is this for? I don't think it makes a lot of difference, but I want to make sure that we are testing the correct things, and it might reduce the total impact to the processing being done.

@mythrocks do you remember why we could not support this out of the box?  In practice I think we could replicate this is essentially doing a lead/lag like operation before the window operation. But I am not really 100% sure on that.

**Comment by mythrocks** [comment]
Sorry for the delay.  This took some recollecting.

TL;DR:  The CUDF primitives I've written do not support negative bounds.  It is doable with some non-trivial effort.

The primitives convert the range values into real row-offset values in the sort-by columns.  This was easy to do in the ROWS case.  In the RANGE case, the maths will be a little confusing.  I'd be delighted to work on it, depending on priority.
